modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/CurrencySnapshotAcceptanceManager.scala

```diff
@@ -126,7 +126,11 @@ object CurrencySnapshotAcceptanceManager {
         messagesToReject
       )
 
-      csi = CurrencySnapshotInfo(transactionsRefs, updatedBalancesByRewards, messagesForContextUpdate.some)
+      csi = CurrencySnapshotInfo(
+        transactionsRefs,
+        updatedBalancesByRewards,
+        Option.when(messagesForContextUpdate.nonEmpty)(messagesForContextUpdate)
```

Comment:
The state of currency messages will appear for the first time when actual messages are processed - SortedMap non empty. From that point on it will always be Some.